paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/CustomerRepositoryKtx.kt

```diff
@@ -0,0 +1,73 @@
+package com.stripe.android.paymentsheet.repositories
+
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * Removes the provided saved payment method alongside any duplicate stored payment methods. This function should be
+ * deleted once an endpoint is stood up to handle detaching and removing duplicates.
+ *
+ * @param customerInfo authentication information that can perform detaching operations.
+ * @param paymentMethodId the id of the payment method to remove and to compare with for stored duplicates
+ *
+ * @return a list of removal results that identify the payment methods that were successfully removed and the payment
+ * methods that failed to be removed.
+ */
+internal suspend fun CustomerRepository.detachPaymentMethodAndDuplicates(
+    customerInfo: CustomerRepository.CustomerInfo,
+    paymentMethodId: String
+): List<PaymentMethodRemovalResult> {
+    val paymentMethods = getPaymentMethods(
+        customerInfo = customerInfo,
+        // We only support removing duplicate cards.
+        types = listOf(PaymentMethod.Type.Card),
+        silentlyFail = false,
+    ).getOrElse {
+        return listOf(
+            PaymentMethodRemovalResult(
+                paymentMethodId = paymentMethodId,
+                result = Result.failure(it)
+            )
+        )
+    }
+
+    val requestedPaymentMethodToRemove = paymentMethods.find { paymentMethod ->
+        paymentMethod.id == paymentMethodId
+    } ?: return listOf()
+
+    val paymentMethodRemovalResults = mutableListOf<PaymentMethodRemovalResult>()
+
+    val paymentMethodsToRemove = paymentMethods.filter { paymentMethod ->
+        paymentMethod.type == PaymentMethod.Type.Card &&
+            paymentMethod.card?.fingerprint == requestedPaymentMethodToRemove.card?.fingerprint
+    }
+
+    paymentMethodsToRemove.forEach { paymentMethod ->
+        val paymentMethodIdToRemove = paymentMethod.id!!
```

Comment:
if paymentMethod.id was null, then we would fail to remove all the payment methods right? Should we instead just fail removing this one? Or could we update this type to not have a null id (seems harder but would be better if we expect it to never be null)?

Reply:
`PaymentMethod` id is a non-nullable type on the `Stripe` API. This was overlooked when designing the client-side representation. I'll fail more gracefully here.

Reply:
sounds good! Do you know if we could also update the client-side representation? This PR looks good to me, but that would be a nice follow up, if it's possible